src/util/screensaver.cpp

```diff
@@ -183,100 +209,122 @@ void ScreenSaverHelper::triggerUserActivity()
     }
     return;
 }
-// Disabling the method with DBus since it seems to be failing on several systems.
-#if 0
-void ScreenSaverHelper::triggerUserActivity()
-{
-    if (!s_sendActivity) {
-        // If the D-Bus method didn't work, let's try the Xlib method.
-        const char* name = ":0.0";
-        Display *display;
-        if (getenv("DISPLAY"))
-            name=getenv("DISPLAY");
-        display=XOpenDisplay(name);
-        XResetScreenSaver(display);
-        XCloseDisplay(display);
-        return;
-    }
-    
-    s_sendActivity = false;
+
+void ScreenSaverHelper::inhibitInternal() {
     if (!QDBusConnection::sessionBus().isConnected()) {
-        qWarning("Cannot connect to the D-Bus session bus.\nTo start it, run:\n"
-                "\teval `dbus-launch --auto-syntax`");
+        qWarning() << "Cannot connect to the D-Bus session bus";
         return;
     }
-    s_sendActivity = false;
-    QString errors;
-    for (int i=0; USERACTIVITY[i][0] != nullptr; i++ ) {
-        QDBusInterface iface(USERACTIVITY[i][0], USERACTIVITY[i][1], USERACTIVITY[i][2], 
-            QDBusConnection::sessionBus());
+
+    // Call uninhibit if the handle / cookie has not been reset
+    if (!s_session_handle.isEmpty() || s_inhibit_cookie > 0) {
+        uninhibit();
+    }
+
+    for (const Inhibitor& inhibitor : kDBusInhibitors) {
+        // Iterate through D-Bus idle inhibitors and select the first available
+        QDBusInterface iface(inhibitor.service,
+                inhibitor.object_path,
+                inhibitor.interface,
+                QDBusConnection::sessionBus());
         if (iface.isValid()) {
-            QDBusReply<void> reply = iface.call(USERACTIVITY[i][3]);
+            QDBusReply<uint32_t> reply = iface.call("Inhibit",
+                    "org.mixxx.Mixxx",
+                    "Mixxx is running");
             if (reply.isValid()) {
-                s_sendActivity = true;
-                break;
+                s_inhibitor = &inhibitor;
+                s_inhibit_cookie = reply.value();
+                s_enabled = true;
+                qDebug() << "Idle inhibitor" << inhibitor.interface << "enabled";
+                return;
             } else {
-                errors = errors +  "\nCall to inhibit for " + USERACTIVITY[i][0] + " failed: " 
-                    + reply.error().message();
+                qWarning() << "Inhibit for"
+                           << inhibitor.interface << "failed: "
+                           << reply.error().message();
             }
-        } 
+        } else {
+            qDebug() << "Interface" << inhibitor.interface << "is not valid";
+        }
     }
```

Comment:
this may cause logspam. It'd be nice if this was only tested once on startup in a follow up PR.

Reply:
I agree, that last part (lines 245 - 247) can be especially spammy and it's not super useful either. Let's just remove it. There's already a message if inhibit was successful / unsuccessful or if no supported inhibitors were found.